lib/sample_rate_converter/Kconfig

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config SAMPLE_RATE_CONVERTER
+	bool "Sample Rate conversion library"
+	select CMSIS_DSP
+	select CMSIS_DSP_FILTERING
+	select TIMING_FUNCTIONS
+	help
+	  Enable the sample rate conversion library. The library uses CMSIS DSP filters to
+	  preserve quality during the conversion. Conversion between 16kHz, 24kHz and 48kHz
+	  frequencies are supported.
+
+if SAMPLE_RATE_CONVERTER
+module = SAMPLE_RATE_CONVERTER
+module-str = Sample Rate Converter
+source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config"
+
+config SAMPLE_RATE_CONVERTER_FILTER_SIMPLE
+	bool "Include the simple sample rate converter filters"
+	help
+	  Includes the simple filters for the sample rate converter. The simple filters uses the
+	  least amount of space and time for the conversion, at the expense of quality.
+
+config SAMPLE_RATE_CONVERTER_FILTER_SMALL
+	bool "Include the small sample rate converter filters"
+	help
+	  Includes the small filters for the sample rate converter. The small filters uses a small
+	  amount of space and time for the conversion, while also giving some low-pass filter
+	  capabilities.
+
+config SAMPLE_RATE_CONVERTER_MAX_FILTER_SIZE
+	int "Maximum number of filter taps the module supports"
+	default 34
+	help
+	  The maximum number of filter taps the sample rate converter supports. Increasing this
+	  number will increase the memory usage of the converter.
```

Comment:
No, it doesn't affect processing time.

I felt it was good to point out the memory usage since this kconfig directly changes the size of the context struct which may often be statically allocated, not dynamically. So basically just to make it clear that this shouldn't mindlessly be set high to "allow for all cases".

Reply:
Will find a way to do this in code and not kconfig